sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_pipeline_and_node_output.yaml

```diff
@@ -0,0 +1,21 @@
+display_name: register_both_output
+type: pipeline
+jobs:
+  parallel_body:
+    type: command
+    inputs:
+      component_in_path:
+        type: uri_file
+        path: https://dprepdata.blob.core.windows.net/demo/Titanic.csv
+    component: ../components/helloworld_component.yml
+    outputs:
+      component_out_path:
+        name: node_output
```

Comment:
we should have validation for output name. Tried to put a invalid data name. raised error in runtime.

https://ml.azure.com/runs/quirky_chin_x7f3pm9rhx?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/sdk/workspaces/sdk-master&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Name is invalid. It can only contain alphanumeric characters, dashes, and underscores, with a limit of 255 characters.

Reply:
related codes are added in Output and NodeOutput. Add test_register_output_without_name_yaml and test_register_output_with_ilvalid_name_sdk to test this.